File: digraph.py
from graph import Graph


class Digraph(Graph):

    def addEdge(self, v, w):
        super().addToList(v, w)

    def toDot(self):
        NEWLINE = '\n'
        sb = "digraph {" + NEWLINE
        sb += "rankdir = LR;" + NEWLINE
        sb += "node [shape = circle];" + NEWLINE
        for v in sorted(self.getVerts()):
            for w in self.getAdj(v):
                sb += v + " -> " + w + NEWLINE
        sb += "}" + NEWLINE
        return sb


if __name__ == "__main__":

    g = Digraph()

    g.addEdge("0", "1")
    g.addEdge("0", "2")
    g.addEdge("2", "1")

    g = Digraph("exemplos/tinyG.txt")

    for v in g.getVerts():
        print(f"{v}: ", end="")
        for w in g.getAdj(v):
            print(f"{w} ", end="")
        print()
    print()
    print(g.toDot())

File: Digraph.java
public class Digraph extends Graph {

  public Digraph() {
    super();
  }

  public Digraph(String filename) {
    super(filename);
  }

  @Override
  public void addEdge(String v, String w) {
    addToList(v, w);
    if (!vertices.contains(v)) {
      vertices.add(v);
      totalVertices++;
    }
    if (!vertices.contains(w)) {
      vertices.add(w);
      totalVertices++;
    }
  }

  @Override
  public String toDot() {
    StringBuilder sb = new StringBuilder();
    sb.append("digraph {" + NEWLINE);
    sb.append("rankdir = LR;" + NEWLINE);
    sb.append("node [shape = circle];" + NEWLINE);
    for (String v : getVerts().stream().sorted().toList())
      for (String w : getAdj(v))
        sb.append("\"" + v + "\"" + " -> " + "\"" + w + "\"" + NEWLINE);
    sb.append("}" + NEWLINE);
    return sb.toString();
  }
}
